### PR TITLE
converts the hide verb to an action button

### PR DESCRIPTION
**Pull request by GDNgit**
## What Does This PR Do
Converts the "Hide" verbs over to action buttons. Done for drones, small simplemobs, and alien larva

## Why It's Good For The Game
Verbs bad and aren't very new player friendly, action buttons are quite simply better in most cases, this being one of them as you'll likely toggle it frequently as like a mouse.

## Testing
Compiled and ran

## Changelog
:cl:
tweak: "Hide" verbs are now action buttons
/:cl:

### CODE REVIEW

File: code/modules/mob/living/carbon/alien/larva/larva.dm [hunk 1]
```diff
@@ -82,3 +90,18 @@
...
+	name = "Hide"
+	desc = "Allows to hide beneath tables or certain items. Toggled on or off."
+	background_icon_state = "bg_alien"
+	check_flags = AB_CHECK_CONSCIOUS
+	button_icon_state = "alien_hide"
+
+/datum/action/innate/alien_larva_hide/Activate()
+	if(owner.layer != ABOVE_NORMAL_TURF_LAYER)
+		owner.layer = ABOVE_NORMAL_TURF_LAYER
+		owner.visible_message("<B>[owner] scurries to the ground!</B>", "<span class='noticealien'>You are now hiding.</span>")
```
Comment (lewcc) [inline code/modules/mob/living/carbon/alien/larva/larva.dm:103]:
```suggestion
		owner.visible_message("<b>[owner] scurries to the ground!</b>", "<span class='noticealien'>You are now hiding.</span>")
```

File: code/modules/mob/living/simple_animal/hide_action.dm [hunk 2]
```diff
@@ -0,0 +1,20 @@
+/datum/action/innate/animal_hide
```
Comment (lewcc) [inline code/modules/mob/living/simple_animal/hide_action.dm:1]: there no good way to incorporate this with the other one?

Comment (GDNgit) [reply code/modules/mob/living/simple_animal/hide_action.dm:1]: not really, uses totally different layers and stuff

File: code/modules/mob/living/simple_animal/hide_action.dm [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+	desc = "Allows to hide beneath tables or certain items. Toggled on or off. While hiding you can fit under unbolted airlocks."
+	check_flags = AB_CHECK_CONSCIOUS
+	button_icon_state = "mouse_gray_sleep"
+
+/datum/action/innate/animal_hide/Activate()
+	var/mob/living/simple_animal/simplemob = owner
+
+	if(simplemob.layer != TURF_LAYER + 0.2)
+		simplemob.layer = TURF_LAYER + 0.2
+		simplemob.visible_message("<B>[src] scurries to the ground!</B>", "<span class=notice'>You are now hiding.</span>")
```
Comment (Sirryan2002) [inline code/modules/mob/living/simple_animal/hide_action.dm:12]:
```suggestion
		simplemob.visible_message("<b>[src] scurries to the ground!</b>", "<span class=notice'>You are now hiding.</span>")
```

File: code/modules/mob/living/simple_animal/hide_action.dm [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+		if(simplemob.pass_door_while_hidden)
+			simplemob.pass_flags |= PASSDOOR
+		return
+
+	simplemob.layer = MOB_LAYER
+	simplemob.visible_message("[src] slowly peeks up from the ground...", "<span class=notice'>You have stopped hiding.</span>")
```
Comment (Sirryan2002) [inline code/modules/mob/living/simple_animal/hide_action.dm:18]: needs a span class please

**Review by Sirryan2002 — COMMENTED** [review]
couple small things

File: code/modules/mob/living/simple_animal/hide_action.dm (hunk 2 above)
Comment (farie82) [reply code/modules/mob/living/simple_animal/hide_action.dm:1]: Can make the new layer into a variable. Then have the larva one be a subtype of this one. The messages could also be variables if needed

**Comment by SteelSlayer** [comment]
@GDNgit Should be good to go after you address reviews and deconflict.

**Comment by farie82** [comment]
Please process the reviews within a week or the PR will be closed. It can be reopened on a later date if this happens.

File: code/modules/mob/living/simple_animal/hide_action.dm (hunk 2 above)
Comment (GDNgit) [reply code/modules/mob/living/simple_animal/hide_action.dm]: it's done

**Review by SteelSlayer — APPROVED** [review]

**Review by Sirryan2002 — APPROVED** [review]

**Pull request merged**